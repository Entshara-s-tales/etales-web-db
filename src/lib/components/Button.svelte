<script lang="ts">
	export let type: "link" | "button" = "button";
</script>

<button type="button" class="text-black rounded">
	<slot>Missing content</slot>
</button>
