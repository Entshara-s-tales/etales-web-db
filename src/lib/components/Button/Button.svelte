<script lang="ts">
	export let type: 'link' | 'button' = 'link';
	export let onClick: () => void = undefined;
</script>

<button on:click={onClick} type="button" class="text-black rounded">
	<slot>Missing content</slot>
</button>
