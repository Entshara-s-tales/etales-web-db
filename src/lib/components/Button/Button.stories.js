import Button from './Button.svelte';

export default {
  title: 'lib/components/Button',
  component: Button,
  argTypes: {},
};

const Template = ({ onClick, ...args }) => ({
  Component: Button,
  props: {
    ...args,
  },
});

export const SampleUsage = Template.bind({});
SampleUsage.args = {
  label: 'Button',
};
