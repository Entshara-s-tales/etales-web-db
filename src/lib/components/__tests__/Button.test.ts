import { render } from "@testing-library/svelte";
import Button from "../Button.svelte";
import TestButton from "./TestButtonWithSlot.svelte";

test("should render with missing content", () => {
	const results = render(Button);

	expect(results.getByText("Missing content")).toReturn;
	expect(results.container).toMatchSnapshot();
});

test("should render", () => {
	const results = render(TestButton);

	expect(() => results.getByText("Test content")).not.toThrow();
	expect(results.container).toMatchSnapshot();
});
