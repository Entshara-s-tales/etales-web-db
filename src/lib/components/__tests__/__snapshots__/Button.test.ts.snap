// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<body>
  <div>
    <button
      class="text-black rounded"
      type="button"
    >
      Test content
    </button>
  </div>
</body>
`;

exports[`should render with missing content 1`] = `
<body>
  <div>
    <button
      class="text-black rounded"
      type="button"
    >
      Missing content
    </button>
  </div>
</body>
`;
